agent: set default caps if gateways not configured

tobsan opened the pull request:
The only way to have the gateways configured from the agent is through
setCapabilities, so if they are not configured, then that call has not
been made, and the default caps have not been applied.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
agent: set default caps if gateways not configured

The only way to have the gateways configured from the agent is through
setCapabilities, so if they are not configured, then that call has not
been made, and the default caps have not been applied.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

breakreturn commented on agent/src/softwarecontaineragent.cpp:
```diff
@@ -297,6 +297,22 @@ bool SoftwareContainerAgent::execute(ContainerID containerID,
+    // from the agent is through setCapabilities - which sets the default caps also.
+    //
+    // If it has not been set, then we use a call without arguments to setCapabilities to set it up,
+    // since we then should get only the default ones.
```
For multi line comments it would be more consistent with the rest of the code to use:
/*
 *
 */

DunderRoffe replied:
I think it is nice to not use that when writing something that should not end up in the API docs as it looks different from the doxygen comments. On the other hand, this might be useful in the API if tweaked a bit?

breakreturn replied:
Do you mean you see a risk that it might be interpreted by someone as a doxygen comment and lead to confusion? AFAIK the most consistent choice now would be to use /* */ based on the rest of the code.

tobsan replied:
FYI this will not be interpreted by doxygen.

Commit:
test_caps: now also test for default capabilities

Testing is done by providing some default capabilities, then not setting
any capabilities, running the helper, and verifying that the default
caps were indeed set.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

breakreturn commented on servicetest/capabilities/test_caps.py:
```diff
@@ -80,13 +83,34 @@ def logfile_path():
+default_cap = Capability("test.cap.default.sample",
+                        [
+                            { "id": "env", "config": ENV_GW_CONFIG }
+                        ])
```
The list is missing one space indentation.

breakreturn commented on servicetest/capabilities/test_caps.py:
```diff
@@ -186,5 +206,32 @@ def test_list_capabilities(self):
+                                  sc.get_bind_dir() +
+                                  "/testhelper.py --test-dir " +
+                                  sc.get_bind_dir() + "/testoutput" +
+                                  " --do-get-env-vars")
```
Indentation looks off

breakreturn commented on servicetest/capabilities/test_caps.py:
```diff
@@ -186,5 +206,32 @@ def test_list_capabilities(self):
+        finally:
+            sc.terminate()
+
+    def test_no_caps_gives_default_only(self):
```
This name is off, it should be something like "test_defaults_are_applied" or something, as it doesn't test any interaction between set caps and default caps, the test result would be the same even if there was some cap set.
